src/op_handlers/shift.rs

```diff
@@ -0,0 +1,75 @@
+use crate::address_operands::{address_operands_read, address_operands_store};
+use crate::value::TaggedValue;
+use crate::{opcode::Opcode, state::VMState};
+use zkevm_opcode_defs::Opcode as Variant;
+use zkevm_opcode_defs::ShiftOpcode;
+
+fn _shl(vm: &mut VMState, opcode: &Opcode) {
+    let (src0_t, src1_t) = address_operands_read(vm, opcode);
+    let (src0, src1) = (src0_t.value, src1_t.value);
+    let shift = (src1.low_u32() % 256) as usize;
+    let res = src0 << shift;
+    if opcode.alters_vm_flags {
+        // Eq is set if result == 0
+        vm.flag_eq |= res.is_zero();
+        // other flags are reset
+        vm.flag_lt_of = false;
+        vm.flag_gt = false;
+    }
+    address_operands_store(vm, opcode, TaggedValue::new_raw_integer(res));
+}
+
+fn _shr(vm: &mut VMState, opcode: &Opcode) {
+    let (src0_t, src1_t) = address_operands_read(vm, opcode);
+    let (src0, src1) = (src0_t.value, src1_t.value);
+    let shift = (src1.low_u32() % 256) as usize;
+    let res = src0 >> shift;
+    if opcode.alters_vm_flags {
+        // Eq is set if result == 0
+        vm.flag_eq |= res.is_zero();
+        // other flags are reset
+        vm.flag_lt_of = false;
+        vm.flag_gt = false;
+    }
+    address_operands_store(vm, opcode, TaggedValue::new_raw_integer(res));
+}
+
+fn _rol(vm: &mut VMState, opcode: &Opcode) {
+    let (src0_t, src1_t) = address_operands_read(vm, opcode);
+    let (src0, src1) = (src0_t.value, src1_t.value);
+    let shift = (src1.low_u32() % 256) as usize;
+    let result = (src0 << shift) | (src0 >> (256 - shift));
```

Comment:
Too bad U256 doesn't implement rotate_left/rotate_right. Looks like this works, nice.

Reply:
Thank you.
At first, I was using `src0.0.rotate_left()` but since it didn't seem to work, I debugged it and it didn't do any rotation (even though I had declared it as mutable).